Core/Specifications/GetRoleByName.cs

```diff
@@ -0,0 +1,13 @@
+﻿using Ardalis.Specification;
+using Core.Entities.RoleEntity;
+
+namespace Core.Specifications
+{
+    public class GetRoleByName : Specification<Role>, ISingleResultSpecification<Role>
+    {
+        public GetRoleByName(string roleName)
+        {
+            Query.Where(r => r.Name == roleName.ToUpper());
+        }
+    }
+}
```

Comment:
Let's aggregate query for each type. Like Specification/<EntityNameClass(It should be static)>.InnerClassNamedByQuery